### DISCUSSION TITLE
I want to hash passwords when saving new user

### DISCUSSION THREAD

**Discussion by anna-osipchyk** [aminalaee/sqladmin#403] [General]
I want to hash passwords when saving new user, but have no idea how to call a method "generate password"

**Comment by anna-osipchyk** [comment]
Apparently I found the answer myself. You just need to override "on_model_change" function, for instance 
   async def on_model_change(self, data: dict, model: Any, is_created: bool) -> None:
        password = data.get("password_hash", None)
        if password:
            data["password_hash"] = generate_password_hash(password)

**Reply by aminalaee** [reply]
Glad you could solve it.
